### DISCUSSION TITLE
Help Needed: Persistent data_gas_consumed and data_gas_price Missing Error During StarkNet Transaction Fee Estimation

### DISCUSSION THREAD

**Discussion by otatataata** [software-mansion/starknet.py#1446] [Q&A]
Hi! I'm encountering a persistent issue when trying to invoke a function on a StarkNet contract using starknet_py. Despite my contract's ABI and calldata being correct, and the node (Infura) seemingly functioning properly, I'm getting the following error:

`Paramètres de run_system pour ExtractResourceFinish:
name: 0x457874726163745265736f7572636546696e697368
calldata: [5, 7809, 1, 1, 4329]
Préparation de l'invocation réussie.
Estimation des frais...
Erreur lors de l'exécution de la transaction: {'data_gas_consumed': ['Missing data for required field.'], 'data_gas_price': ['Missing data for required field.']}`

I've tried both manual fee estimation and the auto_estimate=True option, but the error persists in both cases. Here's what I've checked so far:

- The ABI is correct and matches the contract.
- The calldata seems to be correct (changing it results in different errors).
- I'm using the latest version of starknet_py.
- The node endpoint is functioning, as I can retrieve account balance and other details without issues.
- The error appears when invoking the run_system function on the contract, which requires a name and calldata. The invocation works, but fails at the fee estimation stage, where the response lacks data_gas_consumed and data_gas_price.

```
async def run_extract_resource_finish():
    try:
        contract = Contract(address=contract_address, abi=run_system_abi, provider=account)
    except Exception as e:
        print(f"Error configuring contract: {e}")
        return

    name = 0x457874726163745265736f7572636546696e697368  
    calldata = [0x5, 0x1e81, 0x1, 0x1, 0x10e9]

    print("Preparing invocation...")
    try:
        prepared_invocation = contract.functions["run_system"].prepare_invoke_v1(name, calldata)
        print("Invocation prepared successfully.")

        print("Estimating fees...")
        estimated_fee = await contract.functions["run_system"].prepare_invoke_v1(name, calldata).estimate_fee()
        print(f"Estimated fees: {estimated_fee.overall_fee} wei")

        max_fee = int(3e13)  # Set a max fee of approximately 3 cents (0.00003 ETH)
        fee_to_use = max(estimated_fee.overall_fee, max_fee)

        print(f"Fee to use: {fee_to_use} wei")

        # Invoke the function
        invocation = await prepared_invocation.invoke(max_fee=fee_to_use)
        await invocation.wait_for_acceptance()
        print("Transaction successful!")
        print(f"Transaction hash: {invocation.hash}")
    except Exception as e:
        print(f"Error during transaction execution: {e}")
```

The issue seems to be there: estimated_fee = await contract.functions["run_system"].prepare_invoke_v1(name, calldata).estimate_fee(), because the print before shows, but not the print after.

 Any insights or suggestions on what might be causing this and how to resolve it would be greatly appreciated!

**Comment by franciszekjob** [comment]
Hey @otatataata ,
please have a look at the fee estimate schema in spec [here](https://github.com/starkware-libs/starknet-specs/blob/e5ce8c6f99672d9ddf85a8e1d6ef8e87169d6e8d/api/starknet_api_openrpc.json#L3345).
It has a couple fields, including `data_gas_consumed` and `data_gas_price` which are all required. As you correctly mentioned, these fields are missing, however this error does not come from starknet.py itself. It looks like the node is returning a response without these fields.
